lib/HLSL/DxilOperations.cpp

```diff
@@ -718,6 +727,15 @@ void OP::RemoveFunction(Function *F) {
   }
 }
 
+OP::OpCodeClass OP::GetOpClassForDxilFunction(const Function *F) {
+  auto iter = m_FunctionToOpClass.find(F);
+  if (iter == m_FunctionToOpClass.end()) {
```

Comment:
That is fine in principle if the initial assertion is correct. More generally, we either expect this to be correct by construction (and we can guard with an advisory assertion), or we think it could be wrong and then it should be a point where we propagate properly of throw an exception. Generally we're assuming that other frontends / passes would conform to our assertions, by the way (so it's OK to make the analysis based on our particular pipeline configurations).